strategy.go

```diff
@@ -2,8 +2,8 @@ package kafka
 
 import "sort"
 
-// strategy encapsulates the client side rebalancing logic
-type strategy interface {
+// Strategy encapsulates the client side rebalancing logic
+type Strategy interface {
```

Comment:
I wonder if we should rename this type now that it's going to be part of the public API. "Strategy" is a bit vague.

What do you think about something like `ConsumerStrategy` / `GroupStrategy` / `GroupBalancer` / etc... ?

Reply:
Makes sense.  `Balancer` is already used for partitioning, so might not want to overload it even though `GroupBalancer` is otherwise a good choice.  What do you think of `AssignmentStrategy`?

Reply:
I think "Assignment" isn't used anywhere in the API right now, so it's makes it harder to intuitively figure out what it is. I was thinking "Group" because it reuses the wording of "GroupID" used in the reader configuration.

Reply:
Okay.  I'd propose `GroupBalancer` then unless you think that causes confusion with the `Balancer` for partitioning.